pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -625,11 +631,15 @@ async def run_repomdrecord_download(name, location_href, downloader):
                     checksum_types[record.type] = record_checksum_type
                     record.checksum_type = record_checksum_type
 
+                    if self.mirror and record.location_base:
+                        raise ValueError(XML_BASE_AND_MIRROR_INCOMPATIBLE_ERR_MSG)
+
                     if not self.mirror and record.type not in types_to_download:
                         continue
```

Comment:
If we have more checks/combinations to validate at some point, maybe it will be worth creating a matrix in one place which can state what is allow and what is not and how to fail. Just a thought.